HW3/P3/sum.cl

```diff
@@ -24,8 +27,9 @@ __kernel void sum_coalesced(__global float* x,
     // You can assume get_local_size(0) is a power of 2.
     //
     // See http://www.nehalemlabs.net/prototype/blog/2014/06/16/parallel-programming-with-opencl-and-python-parallel-reduce/
-    for (;;) { // YOUR CODE HERE
-        ; // YOUR CODE HERE
+    int local_size = get_local_size(0);
+    for (int offset = (local_size >> 1); offset > 0; offset >>= 1) {
+        fast[local_id] += fast[local_id + offset];
```

Comment:
You are missing an if statement here that verifies that local_id < offset.
This might be the reason why you are limited to 32 threads in a work group.